src/highrl/utils/teacher_checker.py

```diff
@@ -34,53 +66,45 @@ def get_path_bfs(
     """Check if there is a valid path in the input segment
 
     Args:
-        obstacles (Obstacles): Obstacles object
-        coords (List[List[int]]): Coordinates defining the segment
-        width (int): Width of the env
-        height (int): Height of the env
+        obstacles (Obstacles): Obstacles in the environemnt
+        env_size (int): Environment size. Note that the environment is
+        assumed to be a square here.
         robot_pos (List[int]): Robot position
         goal_pos (List[int]): Goal position
         omit_first_four (bool): Whether to ignore the first four obstacles which usually
-        represent the boarder of the env
+        represent the boarder of the env. Defaults to True.
 
     Returns:
-        bool: flag whether there exists a path
+        Tuple[bool, List[Position[int]]]: Flag whether there exists a path, The points along the
+        generated path.
     """
     delta_x = [-1, -1, -1, 0, 0, 1, 1, 1]
     delta_y = [-1, 0, 1, -1, 1, -1, 0, 1]
     num_dirs = 8
 
     # Intializing empty map
-    # "." represents an empty space in the map
-    env_map = []
-    parent_pos = []
-    for x_pos in range(env_size + 1):
-        current = []
-        current_parents = []
-        for y_pos in range(env_size + 1):
-            current.append(".")
-            current_parents.append(Position[int](-1, -1))
-        env_map.append(current)
-        parent_pos.append(current_parents)
-
-    # Filling obstacles spaces with "X" as means
-    # of representing occupancy in the map
-    for i, obstacle in enumerate(obstacles):
-        if omit_first_four and i < 4:
-            continue
-        points = obstacle.get_grid_points()
-        for point in points:
-            x_pos, y_pos = point
-            env_map[x_pos][y_pos] = "X"
+    # False represents an empty space in the map
+    first_time = time.time()
+    default_pos = Position[int](-1, -1)
+    env_map = np.full((env_size + 1, env_size + 1), False)
+    
+    # Parent position stores the parent cell for each cell in the map
+    # All of its values are initialized as (-1, -1) to ensure that each
+    # point in the generated path MUST have a parent position.
+    parent_pos = [
+        [default_pos for _ in range(env_size + 1)] for _ in range(env_size + 1)
+    ]
+    _LOG.warning("Initialized map in %.2f seconds", time.time() - first_time)
+
     # Breadth first search till you either find the
     # goal or you stop (aka. no valid in this map)
     # Intializing the queue with the robot position
-    queue = [robot_pos]
-    env_map[robot_pos.x][robot_pos.y] = "X"
+    queue = deque([robot_pos], maxlen=env_size)
+    env_map[robot_pos.x][robot_pos.y] = True
```

Comment:
Is there a certain reason why you changed the convention from "." represents an empty space and "x" represents occupancy to the True and False labeling used in this PR?

Reply:
Yes, is the **space**. "X" takes 8 bits (1 byte), yet `False` or `True` take only 1 bit. Hence, 8x faster comparisons and 8x less storage.

Reply:
Noted